src/main/java/htsjdk/samtools/BinaryBAMShardIndexWriter.java

```diff
@@ -0,0 +1,158 @@
+package htsjdk.samtools;
+
+import htsjdk.samtools.util.BinaryCodec;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+
+/**
+ * Only writes header for the first reference
+ */
+public class BinaryBAMShardIndexWriter implements BAMIndexWriter {
+  private final BinaryCodec codec;
+
+  /**
+   * @param nRef Number of reference sequences. If zero is passed then header is not written.
+   * This is useful in sharded writing as we only want the header written for the first shard.
+   * 
+   * @param output BAM index output stream.  This stream will be closed when BinaryBAMIndexWriter.close() is called.
+   */
+  public BinaryBAMShardIndexWriter(final int nRef, final OutputStream output) {
+      try {
+          codec = new BinaryCodec(output);
+          if (nRef > 0) {
+            writeHeader(nRef);
+          }
+      } catch (final Exception e) {
+          throw new SAMException("Exception opening output stream", e);
+      }
+  }
+
+  /**
+   * Write this content as binary output
+   */
+  @Override
+  public void writeReference(final BAMIndexContent content) {
+
+      if (content == null) {
+          writeNullContent();
+          return;
+      }
+
+      // write bins
+
+      final BAMIndexContent.BinList bins = content.getBins();
+      final int size = bins == null ? 0 : content.getNumberOfNonNullBins();
+
+      if (size == 0) {
+          writeNullContent();
+          return;
+      }
+
+      //final List<Chunk> chunks = content.getMetaData() == null ? null
+      //        : content.getMetaData().getMetaDataChunks();
+      final BAMIndexMetaData metaData = content.getMetaData();
+
+      codec.writeInt(size + ((metaData != null)? 1 : 0 ));
+      // codec.writeInt(size);
+      for (final Bin bin : bins) {   // note, bins will always be sorted
+          if (bin.getBinNumber() == GenomicIndexUtil.MAX_BINS)
+              continue;
+          writeBin(bin);
+      }
+
+      // write metadata "bin" and chunks        
+      if (metaData != null)
+          writeChunkMetaData(metaData);
+
+      // write linear index
+
+      final LinearIndex linearIndex = content.getLinearIndex();
+      final long[] entries = linearIndex == null ? null : linearIndex.getIndexEntries();
+      final int indexStart = linearIndex == null ? 0 : linearIndex.getIndexStart();
+      final int n_intv = entries == null ? indexStart : entries.length + indexStart;
```

Comment:
Does `n_intv` mean the nth interval?  If so, maybe something more descriptive like `nthIntervalFromStart` or `indexOffsetFromStart` might communicate quicker and easier.  Again we are talking about references so maybe tying that in into the variable name would allow for easier elucidation for the need of this variable.